_source/_posts/2023-07-27-spring-graphql-react.md

```diff
@@ -0,0 +1,1794 @@
+---
+layout: blog_post
+title: "How to build a GraphQL API with Spring for GraphQL"
+author: jimena-garbarino
+by: contractor
+communities: [security,java,javascript]
+description: "A step-by-step guide for building a secured GraphQL API with Spring Boot and Auth0 authentication on React"
+tags: [java, javascript]
+tweets:
+- ""
+- ""
+- ""
+image:
+type: awareness
+---
+
+GraphQL is a query language for APIs and a runtime that allows the API consumer to get exactly the required information instead of the server exclusively controlling the contents of the response. While some REST API implementations require loading the references of a resource from multiple URLs, GraphQL APIs can follow references between related objects and return them in a single request.
+
+This step-by-step guide demonstrates how to build a GraphQL API with Spring Boot and Spring for GraphQL, for querying a sample dataset of related companies, persons, and properties, seeded to a Neo4j database. It also demonstrates how to build a React client with Next.js and MUI Datagrid to consume the API. Both the client and the server are secured with Auth0 authentication using Okta Spring Boot Starter for the server and Auth0 React SDK for the client.
+
+{% img blog/spring-graphql-react/logos.png alt:"Spring, GraphQL and React" width:"700" %}{: .center-image }
+
+> This example was created with the following tools and services:
+> - [Node.js v18.16.1](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)
+> - [npm 9.5.1](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)
+> - [Java OpenJDK 17](https://jdk.java.net/java-se-ri/17)
+> - [Docker 24.0.2](https://docs.docker.com/desktop/)
+> - [Auth0 account](https://auth0.com/signup)
+> - [Auth0 CLI 1.0.0](https://github.com/auth0/auth0-cli#installation)
+> - [HTTPie 3.2.2](https://httpie.io/)
+> - [Next.js 13.4.19](https://nextjs.org/)
+
+{% include toc.md %}
+
+## Build a GraphQL API with Spring for GraphQL
```

Comment:
I think we should add a section before this that shows how to get started quickly. That is, clone the repo, configure it for Auth0, and run it. If the repo's README has everything, you could condense it to a single paragraph. Or you could follow something similar to what I did in my Angular and Spring Boot post. https://auth0.com/blog/spring-boot-angular-crud/

That way, when this tutorial stops working in a year or two (because the Spring Boot version is no longer available on start.spring.io), people can still see the app running.

Reply:
I was thinking the same while writing the post, I like the structure of the angular post. For this time, I just added a note below the logos image, linking to the repository.